dm/pkg/utils/db.go

```diff
@@ -132,37 +128,20 @@ func GetSlaveServerID(ctx context.Context, db *sql.DB) (map[uint32]struct{}, err
 		+------------+-----------+------+-----------+
 	*/
 
-	var (
-		serverID  sql.NullInt64
-		host      sql.NullString
-		port      sql.NullInt64
-		masterID  sql.NullInt64
-		slaveUUID sql.NullString
-	)
 	serverIDs := make(map[uint32]struct{})
-	for rows.Next() {
-		if len(rowColumns) == 5 {
-			err = rows.Scan(&serverID, &host, &port, &masterID, &slaveUUID)
-		} else {
-			err = rows.Scan(&serverID, &host, &port, &masterID)
-		}
+	var rowsResult []string
+	rowsResult, err = export.GetSpecifiedColumnValueAndClose(rows, "Server_id")
+	if err != nil {
+		return nil, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+	}
+	for _, serverID := range rowsResult {
+		// serverID will not be null
+		serverIDInt, err := strconv.Atoi(serverID)
```

Comment:
may raise error when int is 32-bit but `serverID` is in the upper-half of uint32 values?

Reply:
You can use https://pkg.go.dev/strconv#ParseUint, don't use int because its range is undetermined